payments/providers/bambora_payform.py

```diff
@@ -288,21 +398,33 @@ def initiate_refund(self, order: Order) -> OrderRefund:
                 _("Cannot refund an order that has not been settled yet (active token)")
             )
 
-        refund_amount = order.total_price
         email = order.customer_email or (
             order.lease.application.email
             if hasattr(order, "lease") and getattr(order.lease, "application", None)
             else None
         )
         if not email:
             raise ValidationError(_("Cannot refund an order that has no email"))
+
+        order_number = f"{order.order_number}-{order_token.created_at.timestamp()}"
+
+        # The refund requires the product IDs that Bambora assigns them
+        payment_details = self.get_payment_details(order_number)
+
+        products = []
+        refund_amount = 0
+
+        for product in payment_details.payment_products:
+            products.append({"id": product.product_id, "count": 1})
+            refund_amount = price_from_fractional_int(product.price)
```

Comment:
Would it make sense to validate that refund_amount calculated from products matches order.total_price?
Not necessarily fail, but log a warning if the amounts do not match?

If, for any reason (bug in Bambora or whatever), the refund_amount is something other than order.total_price, then terminate_lease() will not be called later, and it would be good to have some trace about what happened.

Reply:
Hmmm right. I thought about it but I didn't come to a conclusion on how would we handle that case. Do we (devs) have to manually check and fix the order? If it has already been registered on VismaPay, what do we do about it?

Probably we should raise an error since we don't expect it to happen, and if it does, then that means trouble.

Reply:
My reasoning for allowing the refund to proceed with a different amount was that Visma Pay is more likely than us correct about the amount that the customer has actually paid, and using the same amount for refund will more likely lead to the result that customer&admins wanted.

But since this is not expected to happen, raising an error is a good default. If this happens often, we definitely want to know about it so we can investigate.